ksql-rest-app/src/test/java/io/confluent/ksql/rest/util/ClusterTerminatorTest.java

```diff
@@ -0,0 +1,301 @@
+    //Given:
+    givenTopicsExistInKafka("K_Foo");
+    givenSinkTopicExists("K_Foo");
+
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("K_Foo"));
+
+    // Then:
+    verify(kafkaTopicClient).deleteTopics(Collections.singletonList("K_Foo"));
+  }
+
+  @Test
+  public void shouldOnlyDeleteExistingTopics() {
+    //Given:
+    givenTopicsExistInKafka("K_Bar");
+    givenSinkTopicExists("K_Foo", "K_Bar");
+
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("K_Foo", "K_Bar"));
+
+    // Then:
+    verify(kafkaTopicClient).deleteTopics(ImmutableList.of("K_Bar"));
+  }
+
+  @Test
+  public void shouldNotDeleteNonSinkTopic() {
+    // Given:
+    givenNonSinkTopicExits("bar");
+
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("bar"));
+
+    // Then:
+    verify(kafkaTopicClient, never()).deleteTopics(Collections.singletonList("bar"));
+  }
+
+  @Test
+  public void shouldNotDeleteNonMatchingCaseSensitiveTopics() {
+    // Given:
+    givenTopicsExistInKafka("K_FOO");
+    givenSinkTopicExists("K_FOO");
+
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("K_Foo"));
+
+    // Then:
+    verify(kafkaTopicClient, times(2)).deleteTopics(Collections.emptyList());
+  }
+
+
+  @Test
+  public void shouldDeleteTopicListWithPattern() {
+    //Given:
+    givenTopicsExistInKafka("K_Fo", "K_Foo", "K_Fooo", "NotMatched");
+    givenSinkTopicsExistInMetastore("K_Fo", "K_Foo", "K_Fooo", "NotMatched");
+
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("K_Fo.*"));
+
+    // Then:
+    verify(kafkaTopicClient).deleteTopics(ImmutableList.of("K_Fooo", "K_Fo", "K_Foo"));
+  }
+
+  @Test
+  public void shouldRemoveNonExistentTopicsOnEachDeleteAttempt() {
+    //Given:
+    givenSinkTopicsExistInMetastore("Foo", "Bar");
+
+    when(kafkaTopicClient.listTopicNames())
+        .thenReturn(ImmutableSet.of("Other", "Foo", "Bar"))
+        .thenReturn(ImmutableSet.of("Other", "Bar"))
+        .thenReturn(ImmutableSet.of("Other"));
+
+    doThrow(KsqlException.class).when(kafkaTopicClient).deleteTopics(any());
+    doNothing().when(kafkaTopicClient).deleteTopics(Collections.emptyList());
+
+    // When:
+    clusterTerminator.terminateCluster(ImmutableList.of("Foo", "Bar"));
+
+    // Then:
+    final InOrder inOrder = Mockito.inOrder(kafkaTopicClient);
+    inOrder.verify(kafkaTopicClient).deleteTopics(ImmutableList.of("Bar", "Foo"));
+    inOrder.verify(kafkaTopicClient).deleteTopics(ImmutableList.of("Bar"));
+    inOrder.verify(kafkaTopicClient).deleteTopics(ImmutableList.of());
+  }
+
+  @Test
+  public void shouldThrowIfCouldNotDeleteTopicListWithPattern() {
+    // Given:
+    givenTopicsExistInKafka("K_Foo");
+    givenSinkTopicExists("K_Foo");
+    doThrow(KsqlException.class)
+        .doThrow(KsqlException.class)
+        .doThrow(KsqlException.class)
+        .doThrow(KsqlException.class)
+        .doThrow(KsqlException.class)
```

Comment:
Where is it specified that the deletion should be attempted five times before giving up? Wondering whether this test should take that value into account rather than hard-coding it in.